calm_adapter/window_generator/window_generator.py

```diff
@@ -0,0 +1,71 @@
+#!/bin/env python3
+
+from datetime import date, timedelta
+import json
+
+import click
+import boto3
+from tqdm import tqdm
+
+
+class WindowGenerator:
+
+    topic_arn = "arn:aws:sns:eu-west-1:760097843905:calm-adapter"
+    role_arn = "arn:aws:iam::760097843905:role/platform-developer"
+
+    def __init__(self, start, end=None):
+        self.start = start
+        self.end = end or start
+        assert self.start <= self.end, "Start window not be after end window"
+        self.client = self.get_sns_client()
+
+    def run(self):
+        for window in tqdm(list(self.windows)):
+            self.send_window_to_sns(window)
+
+    @property
+    def windows(self):
+        while self.start <= self.end:
+            yield self.start
+            self.start += timedelta(days=1)
+
+    def send_window_to_sns(self, window):
+        msg = json.dumps({"date": window.isoformat()})
+        self.client.publish(
+            TopicArn=self.topic_arn, Message=msg, Subject=f"Window sent by {__file__}"
+        )
+
+    def get_sns_client(self):
+        sts = boto3.client("sts")
+        role = sts.assume_role(
+            RoleArn=self.role_arn, RoleSessionName="AssumeRoleSession1"
+        )
+        credentials = role["Credentials"]
+        return boto3.client(
+            "sns",
+            aws_access_key_id=credentials["AccessKeyId"],
+            aws_secret_access_key=credentials["SecretAccessKey"],
+            aws_session_token=credentials["SessionToken"],
+        )
+
+
+@click.group()
+def cli():
+    pass
+
+
+@cli.command()
+def full_ingest():
+    start = date(2015, 9, 30) # First date returning any records
```

Comment:
I reckon it might be better to have this as a
```python3
@click.option('--start', default=date(2015, 9, 30))
```
so it could be overridden without modifying this tool. Feels more like config than code.

Reply:
If you're going to do this, change the name as well.

It’s either a full ingest – every record created, which is every record created after 30 September 2015, which is a static date – or it’s a window which defaults to getting the entire set, but can be restricted further.

Reply:
Thanks for suggestions, I'm gonna add a new command for this and keep the full ingest